packages/flutter/test/widgets/heroes_test.dart

```diff
@@ -3159,4 +3183,23 @@ Future<void> main() async {
       testBinding.window.clearAllTestValues();
     },
   );
+
+  testWidgets('Inserting overlay below an existing one', (WidgetTester tester) async {
+    final GlobalKey<NavigatorState> navigatorKey = GlobalKey<NavigatorState>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        navigatorKey: navigatorKey,
+        home: const Scaffold(body: Hero(tag: 'a', child: Placeholder())),
+      )
+    );
+    await tester.pumpAndSettle();
+
+    final AdditionalOverlayRoute route = AdditionalOverlayRoute();
+    navigatorKey.currentState!.push(route);
+    await tester.pump();
+
+    final OverlayState overlay = navigatorKey.currentState!.overlay!;
+    expect(overlay.entries.indexOf(route.overlayEntry), overlay.entries.length - 1);
```

Comment:
Ideally I'd want to check the position of hero's flight `OverlayEntry` however it seems that exposing it would be a pain in the ass. I changed the assertion to verify that *some* entry is inserted between route entries. I think this should be enough.